### PR TITLE
Add sniped to use with tqdm

### PR DESCRIPTION
**Pull request by tito21**
Adds a small code snipped to README.md to show how to use tqdm to add a progress bar for parallel computation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -91,6 +91,29 @@ S, val, vec = parallel_structure_tensor_analysis(data, sigma, rho, devices=4*['c
...
+        if total is not None:
+            self.total = total
+        return self.update(1)
+
+with STETqdm() as t:
+    S, val, vec = parallel_structure_tensor_analysis(
+        image_reduced,
+        sigma=0.1,
+        rho=0.2,
+        truncate=4.0,
```
Comment (Skielex) [inline README.md]: `truncate=4.0` is default, so let's remove this line.

File: README.md [hunk 2]
```diff
@@ -91,6 +91,29 @@ S, val, vec = parallel_structure_tensor_analysis(data, sigma, rho, devices=4*['c
 
 The ideal block size depends on the `sigma` and `rho`, the devices, and the memory available for the devices. Usually values between 100 and 400 work well. **If you encounter out-of-memory errors, try reducing the block size and/or the number of processes.**
 
+You can use the following snipped to get a progress bar (with [tqdm](https://github.com/tqdm/tqdm#hooks-and-callbacks))
```
Comment (Skielex) [inline README.md]: Let's add punctuation to the end of the sentence.
`(with [tqdm](https://github.com/tqdm/tqdm#hooks-and-callbacks)).` is fine.

File: README.md [hunk 3]
```diff
@@ -91,6 +91,29 @@ S, val, vec = parallel_structure_tensor_analysis(data, sigma, rho, devices=4*['c
...
+``` python
+class STETqdm(tqdm):
+    def update_with_total(self, n=1, total=None):
+        if total is not None:
+            self.total = total
+        return self.update(1)
+
+with STETqdm() as t:
+    S, val, vec = parallel_structure_tensor_analysis(
+        image_reduced,
```
Comment (Skielex) [inline README.md]: Let's stick with `data` or `volume` instead of `image_reduced`.

File: README.md [hunk 4]
```diff
@@ -91,6 +91,29 @@ S, val, vec = parallel_structure_tensor_analysis(data, sigma, rho, devices=4*['c
 
 The ideal block size depends on the `sigma` and `rho`, the devices, and the memory available for the devices. Usually values between 100 and 400 work well. **If you encounter out-of-memory errors, try reducing the block size and/or the number of processes.**
 
+You can use the following snipped to get a progress bar (with [tqdm](https://github.com/tqdm/tqdm#hooks-and-callbacks))
+
+``` python
+class STETqdm(tqdm):
```
Comment (Skielex) [inline README.md]: How about calling the class `TqdmTotal` which aligns more with [their examples](https://github.com/tqdm/tqdm?tab=readme-ov-file#hooks-and-callbacks)?

File: README.md [hunk 5]
```diff
@@ -91,6 +91,29 @@ S, val, vec = parallel_structure_tensor_analysis(data, sigma, rho, devices=4*['c
...
+        return self.update(1)
+
+with STETqdm() as t:
+    S, val, vec = parallel_structure_tensor_analysis(
+        image_reduced,
+        sigma=0.1,
+        rho=0.2,
+        truncate=4.0,
+        devices=7 * ["cpu"],
+        structure_tensor=None,
```
Comment (Skielex) [inline README.md]: `structure_tensor=None` is the default, so let's remove this line.

**Review by Skielex — COMMENTED** [review]
Thanks for the adding the example, much appreciated :)

**Comment by tito21** [comment]
I updated the snipped following your recommendations. Feel free to suggest something else.

**Pull request merged**